# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.h
# Commit: c428c3ad7725
# Full Hash: c428c3ad7725593f33b4ea2da9851be31be0c1df
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-11-28 15:41:50
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r ce553434720c -r c428c3ad7725 dom/base/Document.h
--- a/dom/base/Document.h	Mon Nov 28 09:45:39 2022 +0000
+++ b/dom/base/Document.h	Mon Nov 28 09:53:06 2022 +0000
@@ -3768,6 +3768,10 @@
   }
   DOMIntersectionObserver& EnsureLazyLoadImageObserver();
 
+  DOMIntersectionObserver& EnsureContentVisibilityObserver();
+  void ObserveForContentVisibility(Element&);
+  void UnobserveForContentVisibility(Element&);
+
   ResizeObserver* GetLastRememberedSizeObserver() {
     return mLastRememberedSizeObserver;
   }
@@ -5136,6 +5140,10 @@
   // Used to measure how effective the lazyload thresholds are.
   RefPtr<DOMIntersectionObserver> mLazyLoadImageObserverViewport;
 
+  // Used for detecting when `content-visibility: auto` elements are near
+  // or far from the viewport.
+  RefPtr<DOMIntersectionObserver> mContentVisibilityObserver;
+
   // ResizeObserver for storing and removing the last remembered size.
   // @see {@link https://drafts.csswg.org/css-sizing-4/#last-remembered}
   RefPtr<ResizeObserver> mLastRememberedSizeObserver;